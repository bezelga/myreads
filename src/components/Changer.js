import React, { Component } from 'react'
import PropTypes from 'prop-types';
import { shelvesDictionary } from '../utils'

class Changer extends Component {
  render () {
    return (
      <div className="book-shelf-changer">
        <select
         defaultValue={this.props.currentBookshelf}
         onChange={(event) => this.props.changeBookStateFunction(event.target.value)}
       >
          <option value="move" disabled>Move to...</option>
          {Object.entries(shelvesDictionary).map(([key, value]) => (
            <option key={key} value={key}>{value}</option>
          ))}
        </select>
      </div>
    )
  }
}

Changer.propTypes = {
  currentBookshelf: PropTypes.string.isRequired,
  changeBookStateFunction: PropTypes.func.isRequired
}

export default Changer
