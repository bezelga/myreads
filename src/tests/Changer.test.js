import React from 'react'
import Changer from '../components/Changer'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'

const changeBookStateFunction = () => () => {}

const changerSample = <Changer currentBookshelf="read" changeBookStateFunction={changeBookStateFunction}/>

it('matches the changer contract with final snapshot', () => {
  const component = renderer.create(changerSample);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('Changer changes state', () => {
  // const changer = shallow(changerSample)

  // changer.find('select').simulate('change', {target: { value : 'read'}});
  // expect(changer.find('select [selected]').val()).to.equal('read');
  // expect(changer.find('select')).toBe("Read");

});
