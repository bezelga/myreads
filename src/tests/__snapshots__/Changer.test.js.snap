// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the changer contract with final snapshot 1`] = `
<div
  className="book-shelf-changer"
>
  <select
    defaultValue="read"
    onChange={[Function]}
  >
    <option
      disabled={true}
      value="move"
    >
      Move to...
    </option>
    <option
      value="currentlyReading"
    >
      Currently Reading
    </option>
    <option
      value="wantToRead"
    >
      Want to Read
    </option>
    <option
      value="read"
    >
      Read
    </option>
    <option
      value="none"
    >
      None
    </option>
  </select>
</div>
`;
